CRM/Core/Page.php

```diff
@@ -252,7 +254,7 @@ public function run() {
     //its time to call the hook.
     CRM_Utils_Hook::alterContent($content, 'page', $pageTemplateFile, $this);
 
-    echo CRM_Utils_System::theme($content, $this->_print);
+    echo CRM_Utils_System::theme($content, (bool) $this->_print);
```

Comment:
This is the one piece of this PR that gives me pause. `CRM_Utils_System::theme` gets delegated to the CMS (Drupal, WP, Joomla, etc.). Are you sure that every variant of this function in every CMS only considers this variable to be a boolean?

Reply:
@colemanw In CRM_Utils_System, the `$print` argument is documented as accepting a bool. I know that the docblocks are often incorrect, but in this instance it seems that only two theme functions exist:

CRM_Utils_System_Base::theme
CRM_Utils_System_Drupal6::theme

In both of these methods $print is only used as a `!$print` condition within if statements. I also cannot see any hooks of relevance in this area of the code. So I think `$print` really is being treated as a boolean.

That said, happy to revert this bit of the PR if that'd give you the confidence to merge the rest.

Reply:
Yea I would have more confidence without this casting (and its twin later down in the diff). I feel like the direction we're moving is to cast arguments via type hints in the function accepting the argument, rather than in the code passing the argument in. That feels like a good pattern because
1. You only have to do it once, when declaring the function, not every time the function is called
2. You do it right above where the argument is used, so it's easier to see the type cast happening, and any silly code in the function expecting the wrong type would be easier to spot

Reply:
That said, I'd feel safer to merge this PR without any casting of that variable (neither when declaring nor calling the function) and if desired a smaller PR could add the casting as a follow-up, which would be easier to review.

Reply:
I've removed the casting.